soc/xtensa/intel_adsp/common/include/intel_adsp_hda.h

```diff
@@ -469,4 +469,26 @@ static inline uint32_t intel_adsp_hda_check_buffer_interrupt(uint32_t base, uint
 	return (*DGCS(base, regblock_size, sid) & DGCS_BSC) == DGCS_BSC;
 }
 
+/**
+ * @brief Set the Sample Container Size (SCS)
+ *
+ * Sample Container Size indicates the container size of the audio samples in local memory
+ * SCS bit must cleared to 0 for 32bit sample size (HD Audio container size)
+ * SCS bit must be set to 1 for non 32bit sample sizes
+ *
+ * @param base Base address of the IP register block
+ * @param regblock_size Register block size
+ * @param sid Stream ID
+ * @param sample_size
+ */
+static inline void intel_adsp_hda_set_sample_container_size(uint32_t base, uint32_t regblock_size,
+							    uint32_t sid, uint32_t sample_size)
+{
+	if (sample_size <= 3) {
+		*DGCS(base, regblock_size, sid) |= DGCS_SCS;
+	} else {
+		*DGCS(base, regblock_size, sid) &= ~DGCS_SCS;
+	}
```

Comment:
do we need braces for the if/else?

Reply:
@ranj063 yes, this is actually required in Zephyr